src/EventTeachingService.cpp

```diff
@@ -448,11 +448,15 @@ Processed EventTeachingService::handleLearnEventIndex(CANFrame *msg)
         // Writes the first four bytes NN & EN only if they have changed.
         byte eventTableNNEN[4];
         module_config->readEvent(index, eventTableNNEN);
-        if (memcmp(eventTableNNEN, &msg->data[1], 4) == 0)
+        if (memcmp(eventTableNNEN, &msg->data[1], 4) != 0)
         {
           module_config->writeEvent(index, &msg->data[1]);
         }
 
+        if (evIndex == 0)  // Not a valid evIndex
+        {
+          evIndex = 1;
```

Comment:
I think I would respond with an error message instead of changing the input data. It might be confusing when a different event variable is changed than what the user exected.